hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/internal/DataSourceInternalWriterHelper.java

```diff
@@ -97,7 +97,6 @@ public void commit(List<WriteStatus> writeStatuses) {
 
   public void abort() {
     LOG.error("Commit " + instantTime + " aborted ");
-    writeClient.rollback(instantTime);
```

Comment:
Remove the rollback here, since there may still exists some task running and will produce new parquet file.

Reply:
So when the commit got rollback.

Reply:
Curious why the cow table does not encounter the dirty data issue.

Reply:
> So when the commit got rollback.

It's safe to remove the rollback here, since we already have eager(rollback any failed write before commit) and lazy (clean will rollback expired failed writes) rollback mechanism. 

> Curious why the cow table does not encounter the dirty data issue. Or maybe it exists, but no test can cover it.

Yes, this bug has nothing to do with tabletype. If we bulk insert into hudi table using datasource v2 and fail some task, we may encounter this bug.

Reply:
The fix makes sense based on the information created.  @stream2000 could you add a test to verify that after bulk insert with DS v2 fails, the commit is left inflight, and a subsequent new writer / transaction rolls back the commit?

Reply:
Will add a test in the next PR.